core/src/main/scala/io/gearpump/util/HistoryMetricsService.scala

```diff
@@ -80,11 +82,26 @@ class HistoryMetricsService(name: String, config: HistoryMetricsConfig) extends
     result.toList
   }
 
-  def commandHandler: Receive = {
+  private var aggregators: Map[String, MetricsAggregator] = Map("" -> new DummyMetricsAggregator)
 
+  def commandHandler: Receive = {
     //path accept syntax ? *, ? will match one char, * will match at least one char
-    case QueryHistoryMetrics(inputPath, readLatest) =>
-      sender ! HistoryMetrics(inputPath, fetchMetricsHistory(inputPath, readLatest))
+    case QueryHistoryMetrics(inputPath, readLatest, aggregatorClazz) =>
+
+       val aggregator = {
+         if (!aggregators.contains(aggregatorClazz)) {
+           val aggregatorTry = Try(Class.forName(aggregatorClazz).newInstance().asInstanceOf[MetricsAggregator])
+           if (aggregatorTry.isSuccess) {
+             aggregators += aggregatorClazz -> aggregatorTry.get
+           } else {
+             LOG.error("Cannot apply aggregator " + aggregatorClazz, aggregatorTry.failed.get)
+             aggregators += aggregatorClazz -> new SkipAllAggregator
+           }
+         }
+         aggregators(aggregatorClazz)
+       }
+
+      sender ! HistoryMetrics(inputPath, aggregator.aggregate(fetchMetricsHistory(inputPath, readLatest)))
```

Comment:
yes. What I'm concerned is whether we are responsive to users. Does user know what fails and why ?